### ISSUE TITLE
[NLL] rejects-valid vec of array usage

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust#46994]
This code compiles without feature(nll):

```
#![feature(nll)]
fn main() {
    let mut m = vec![[0; 1]; 1];
    m[0][0] = 1;
}
```

With feature(nll) it gives:

```
error[E0597]: `m` does not live long enough
 --> ...\test.rs:4:5
  |
4 |     m[0][0] = 1;
  |     ^ borrowed value does not live long enough
5 | }
  |  - borrowed value only lives until here
  |
  = note: borrowed value must be valid for lifetime '_#2r...
```

**Comment by leonardo-m** [comment]
Perhaps another example of the same problem:

```
#![feature(nll)]
fn main() {
    let mut a = [0u8; 1];
    a[0] = 0.to_string().as_bytes()[0];
}
```

Gives:

```
error[E0597]: borrowed value does not live long enough
 --> ...\test.rs:4:12
  |
4 |     a[0] = 0.to_string().as_bytes()[0];
  |            ^^^^^^^^^^^^^               - temporary value only lives until here
  |            |
  |            temporary value does not live long enough
  |
  = note: borrowed value must be valid for lifetime '_#3r...
```

**Comment by arielb1** [comment]
This is fixed on my branch - #46984 - so I'll close it as a duplicate of #46875.

**Comment by CryZe** [comment]
This seems to be fixed now.
